bootstrap/integration/hooks_integration_test.go

```diff
@@ -488,3 +490,85 @@ func TestPreExitHooksFireAfterCancel(t *testing.T) {
 
 	tester.CheckMocks(t)
 }
+
+func TestPolyglotScriptHooksCanBeRun(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("script hooks aren't supported on windows")
+	}
+
+	path, err := exec.LookPath("ruby")
+	if err != nil {
+		t.Fatal("error finding path to ruby executable: %w", err)
+	}
+
+	if path == "" {
+		t.Fatal("ruby not found in $PATH. This test requires ruby to be installed on the host")
+	}
+
+	defer experiments.EnableWithUndo(experiments.PolyglotHooks)()
+
+	tester, err := NewBootstrapTester()
+	if err != nil {
+		t.Fatalf("NewBootstrapTester() error = %v", err)
+	}
+	defer tester.Close()
+
+	filename := "environment"
+	script := []string{
+		"#!/usr/bin/env ruby",
+		`puts "ohai, it's ruby!"`,
+	}
+
+	if err := os.WriteFile(filepath.Join(tester.HooksDir, filename), []byte(strings.Join(script, "\n")), 0755); err != nil {
+		t.Fatalf("os.WriteFile(%q, script, 0755 = %v", filename, err)
+	}
+
+	tester.RunAndCheck(t)
+
+	if !strings.Contains(tester.Output, "ohai, it's ruby!") {
+		t.Fatalf("tester.Output %s does not contain expected output: %q", tester.Output, "ohai, it's ruby!")
```

Comment:
Format this with %q as well? (tester.Output being an empty string seems like a likely test failure mode)